### ISSUE TITLE
Refactor crate

### ISSUE THREAD

**Issue by Nilirad** [Nilirad/bevy_prototype_lyon#39]
Currently the crate isn't at its optimal state. A good refactoring may fix.

**Comment by Dummyc0m** [comment]
The system ShapePlugin includes despawns the previous entity and spawns a new one with sprite bundle. As a result, the newly spawned entity's id is unobtainable. This seems to be a usability regression?

**Comment by Nilirad** [comment]
I don't think so. Unless `commands.spawn` returns the newly created entity id (and it doesn't), nothing gets lost.

What I did is, instead of creating directly a `SpriteBundle`, the user creates a `ShapeDescriptor` by calling the `draw` method. Then, in the app stage after "update" that I created, the system `shapesprite_maker` creates the sprite bundle and despawns the descriptor.

This approach enables a user to draw shapes without managing manually tessellators and meshes.
